components/kyma-environment-broker/internal/process/deprovisioning/btp_operator_cleanup.go

```diff
@@ -0,0 +1,195 @@
+	}
+	return operation, 0, nil
+}
+
+func (s *BTPOperatorCleanupStep) deleteServiceBindingsAndInstances(k8sClient client.Client, log logrus.FieldLogger) error {
+	namespaces := corev1.NamespaceList{}
+	if err := k8sClient.List(context.Background(), &namespaces); err != nil {
+		return err
+	}
+	requeue := s.deleteResource(k8sClient, namespaces, schema.GroupVersionKind{Group: btpOperatorGroup, Version: btpOperatorApiVer, Kind: btpOperatorBinding}, log)
+	requeue = requeue || s.deleteResource(k8sClient, namespaces, schema.GroupVersionKind{Group: btpOperatorGroup, Version: btpOperatorApiVer, Kind: btpOperatorServiceInstance}, log)
+	if requeue {
+		return fmt.Errorf("waiting for resources to be deleted")
+	}
+	return nil
+}
+
+func (s *BTPOperatorCleanupStep) removeFinalizers(k8sClient client.Client, namespaces corev1.NamespaceList, gvk schema.GroupVersionKind, log logrus.FieldLogger) {
+	listGvk := gvk
+	listGvk.Kind = gvk.Kind + "List"
+	for _, ns := range namespaces.Items {
+		list := &unstructured.UnstructuredList{}
+		list.SetGroupVersionKind(listGvk)
+		if err := k8sClient.List(context.Background(), list, client.InNamespace(ns.Name)); err != nil {
+			log.Errorf("failed listing resource %v in namespace %v", gvk, ns.Name)
+		}
+		for _, r := range list.Items {
+			r.SetFinalizers([]string{})
+			if err := k8sClient.Update(context.Background(), &r); err != nil {
+				log.Errorf("failed remove finalizer for resource %v: %v/%v", gvk, r.GetNamespace(), r.GetName())
+			}
+		}
+	}
+}
+
+func (s *BTPOperatorCleanupStep) deleteResource(k8sClient client.Client, namespaces corev1.NamespaceList, gvk schema.GroupVersionKind, log logrus.FieldLogger) (requeue bool) {
+	listGvk := gvk
+	listGvk.Kind = gvk.Kind + "List"
+	stillExistingCount := 0
+	for _, ns := range namespaces.Items {
+		list := &unstructured.UnstructuredList{}
+		list.SetGroupVersionKind(listGvk)
+		if err := k8sClient.List(context.Background(), list, client.InNamespace(ns.Name)); err != nil {
+			log.Errorf("failed listing resource %v in namespace %v", gvk, ns.Name)
+			if k8serrors2.IsNoMatchError(err) {
+				// CRD doesn't exist anymore
+				return false
+			}
+			requeue = true
+		}
+		stillExistingCount += len(list.Items)
+	}
+	if stillExistingCount == 0 {
+		return
+	}
+	requeue = true
+	for _, ns := range namespaces.Items {
+		obj := &unstructured.Unstructured{}
+		obj.SetGroupVersionKind(gvk)
+		if err := k8sClient.DeleteAllOf(context.Background(), obj, client.InNamespace(ns.Name)); err != nil {
+			log.Errorf("failed deleting resources %v in namespace %v", gvk, ns.Name)
+		}
+	}
+	return
+}
+
+func (s *BTPOperatorCleanupStep) isNotFoundErr(err error) bool {
+	return strings.Contains(err.Error(), "not found")
+}
+
+func (s *BTPOperatorCleanupStep) retryErrors(op internal.DeprovisioningOperation, err error, log logrus.FieldLogger, msg string) (internal.DeprovisioningOperation, time.Duration, error) {
```

Comment:
I don't see any calls of this method, seems overlooked.

Reply:
Thanks, you are correct. I indeed forgot to replace `handleError` calls with `retryErrors`. Addressed now and also renamed to `retryOnError` as I think it better describes the function.